futures-util/src/stream/mod.rs

```diff
@@ -552,6 +555,63 @@ pub trait StreamExt: Stream {
         ForEach::new(self, f)
     }
 
+    /// Runs this stream to completion, executing the provided asynchronous
+    /// closure for each element on the stream concurrently as elements become
+    /// available.
+    ///
+    /// This is similar to [`StreamExt::for_each`], but the futures
+    /// produced by the closure are run concurrently.
+    ///
+    /// The closure provided will be called for each item this stream produces,
+    /// yielding a future. That future will then be executed to completion
+    /// concurrently with the other futures produced by the closure.
```

Comment:
@Nemo157 Single task, not thread (the task may run on different threads depending on the whims of the executor) 😁

I would have understood the current phrasing, however I suppose it can't hurt to make the distinction a bit clearer

Reply:
+1 to clarifying "concurrent, not parallel".

Reply:
@MajorBreakfast "single thread _at a time_" is what I meant.